producer/writer/message_writer.go

```diff
@@ -402,6 +415,17 @@ func (w *messageWriterImpl) scanBatchWithLock(
 			}
 			continue
 		}
+		// If the message exceeded its allowed ttl of the consumer service,
+		// remove it from the buffer.
+		if w.messageTTLNanos > 0 && m.InitNanos()+w.messageTTLNanos <= nowNanos {
+			// There is a chance the message was acked right before the ack is
+			// called, in which case just remove it from the queue.
+			if w.Ack(m.Metadata()) {
```

Comment:
nit: seems you can just fall through here, i.e.,
```
if w.messageTTLNanos > 0 && m.InitNanos()+w.messageTTLNanos <= nowNanos {
  w.Ack(m.Metadata())
  w.m.messageDroppedTTLExpire.Inc(1)
}
if m.IsAcked() {
  w.removeFromQueueWithLock(e, m)
  w.m.messageAcked.Inc(1)
  continue
}
```
?

Reply:
I tried that, but feels cleaner to me if all the if condition has a continue/break in the end versus some falls through and some not.